.gitignore

```diff
@@ -1,2 +1,8 @@
-/compile_commands.json
+#
+# prevent in source build!
+#
+/CMakeCache.txt
+/CMakeFiles/
+/CMakeUserPresets.json
 /build
+/stagedir
```

Comment:
What is this directory for?

Reply:
to use for `DESTDIR=$PWD/stagedir ninja -C build install` instead to `/tmp`

Reply:
see i.e.: https://github.com/beman-project/execution26/blob/main/cmake/CMakeGenericPresets.json

Reply:
Thanks for the context!